src/jwt/rsa-verifier.js

```diff
@@ -0,0 +1,76 @@
+/*
+Based on the work of Tom Wu
+http://www-cs-students.stanford.edu/~tjw/jsbn/
+http://www-cs-students.stanford.edu/~tjw/jsbn/LICENSE
+*/
+
+import {BigInteger} from 'jsbn';
+import SHA256 from 'crypto-js/sha256';
+
+var DigestInfoHead = {
+  sha1: '3021300906052b0e03021a05000414',
+  sha224: '302d300d06096086480165030402040500041c',
+  sha256: '3031300d060960864801650304020105000420',
+  sha384: '3041300d060960864801650304020205000430',
+  sha512: '3051300d060960864801650304020305000440',
+  md2: '3020300c06082a864886f70d020205000410',
+  md5: '3020300c06082a864886f70d020505000410',
+  ripemd160: '3021300906052b2403020105000414',
+};
+
+var DigestAlgs = {
+  sha256: SHA256,
+};
+
+function RSAVerifier(modulus, exp) {
+  this.n = null;
+  this.e = 0;
+
+  if (modulus != null && exp != null && modulus.length > 0 && exp.length > 0) {
+    this.n = new BigInteger(modulus, 16);
+    this.e = parseInt(exp, 16);
+  } else {
+    throw new Error('Invalid key data');
```

Comment:
Is this error message used somewhere?

Reply:
Do you mean when this is thrown? If so, is when the key arguments are invalid and lead to an impossible key instantiation.

Reply:
This _message_ is not used. However, the error needs to be thrown in order to catch it later (on this same promise chain) and reject the Promise with the same "key not found" exception when the key id is missing or the request failed. Avoids duplicating the custom exception instantiation and Promise rejection.